src/test/java/org/jitsi/meet/test/DesktopSharingTest.java

```diff
@@ -37,150 +29,115 @@
-                pb.redirectError(ProcessBuilder.Redirect.INHERIT);
-                Process p = pb.start();
-
-                waitStartSignal.countDown();
-
-                p.waitFor();
-                print("Script ended execution.");
-                waitEndSignal.countDown();
-            }
-            catch (IOException | InterruptedException e)
-            {
-                hookException = e;
-                waitStartSignal.countDown();
-            }
-        }).start();
-
-        final Participant owner = getParticipant1();
-
-        // now lets wait starting or error on startup
-        try
-        {
-            waitStartSignal.await(2, TimeUnit.SECONDS);
-            if (hookException != null)
-            {
-                hookException.printStackTrace();
-                fail("Error executing hook script:"
-                    + hookException.getMessage());
-            }
-        }
-        catch (InterruptedException e)
-        {
-            e.printStackTrace();
-        }
+    /**
+     * Starts desktop sharing.
+     */
+    private void startDesktopSharing()
+    {
+        ((WebParticipant)getParticipant2())
+            .getToolbar().clickDesktopSharingButton();
+    }
 
-        // let's wait some time the user to joins
-        TestUtils.waitForBoolean(
-            owner.getDriver(),
-            "return (APP.conference.membersCount == 2);",
-            25);
-        Participant ownerParticipant = getParticipant1();
-        ownerParticipant.waitForIceConnected();
-        ownerParticipant.waitForSendReceiveData();
-        ownerParticipant.waitForRemoteStreams(1);
-
-        // now lets check whether his stream is screen
-        String remoteParticipantID = owner.getDriver()
-            .findElement(By.xpath("//span[starts-with(@id, 'participant_') " +
-                " and contains(@class,'videocontainer')]")).getAttribute("id");
-        remoteParticipantID
-            = remoteParticipantID.replaceAll("participant_", "");
-
-        final String expectedResult = "desktop";
-        // holds the last retrieved value for the remote type
-        final Object[] remoteVideoType = new Object[1];
-        try
-        {
-            final String scriptToExecute = "return APP.UI.getRemoteVideoType('"
-                + remoteParticipantID + "');";
-            (new WebDriverWait(owner.getDriver(), 5))
-                .until((ExpectedCondition<Boolean>) d -> {
-                    Object res = owner.executeScript(scriptToExecute);
-                    remoteVideoType[0] = res;
-
-                    return res != null && res.equals(expectedResult);
-                });
-        }
-        catch (TimeoutException e)
-        {
-            assertEquals(
-                expectedResult, remoteVideoType[0],
-                "Wrong video type, maybe desktop sharing didn't work");
-        }
+    /**
+     * Stops desktop sharing.
+     */
+    private void stopDesktopSharing()
```

Comment:
startDesktopSharing and stopDesktopSharing are exactly the same? Is this intentional?

Reply:
Kind of. After the refactoring I end up implementing a method that handles both cases. Maybe I should remove the start/stopDesktopSharing and use the new method directly?

Reply:
I'm merging anyway and replaced the two methods with "toggleDesktopSharing"